cinema/views.py

```diff
@@ -48,6 +73,32 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
```

Comment:
First will be good to solve N+1 problem
So you can wrap your code in `if self.actions == "list"` also
And don't forget about retrieve

Reply:
You should fix n+1 with movie and cinema_hall